Installation issue: strelka

### Steps to reproduce the issue

```console
$ spack install strelka target=x86_64
==> Installing strelka
==> No binary for strelka found: installing from source
==> Using cached archive: /software/spack/var/spack/cache/_source-cache/archive/27/27415f7c14f92e0a6b80416283a0707daed121b8a3854196872981d132f1496b.tar.bz2
==> strelka: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' <options>

1 error found in build log:
     12    -- Detecting C compile features - done
     13    -- Detecting CXX compiler ABI info
     14    -- Detecting CXX compiler ABI info - done
     15    -- Check for working CXX compiler: /software/spack/lib/spack/env/gcc
           /g++ - skipped
     16    -- Detecting CXX compile features
     17    -- Detecting CXX compile features - done
  >> 18    CMake Error at CMakeLists.txt:134 (if):
     19      if given arguments:
     20    
     21        "VERSION_LESS" "2.4"
     22    
     23      Unknown arguments specified
     24
```

### Information on your system

```console
$ spack debug report
* **Spack:** 0.15.4-1213-d3d9807
* **Python:** 2.7.5
* **Platform:** linux-centos7-broadwell
```

### Additional information
Won't upload the files, as it takes a good amount of effort to extract them from an air gapped environment, and probably do not bring anything interesting here.

Checking which CMakeLists.txt file is the culprit leads to `spack-src/CMakeLists.txt`, and more especially the line:
`if (${PYTHON_VERSION_STRING} VERSION_LESS ${MINIMUM_PYTHON_VERSION})`
The error message seems to indicate that `${PYTHON_VERSION_STRING}` is empty.

Moreover, the [CMakeLists file](https://github.com/Illumina/strelka/blob/v2.9.x/CMakeLists.txt) refers to "# Find python2 interpreter", while `spack spec strelka%gcc@4.8.5 target=x86_64` refers to
`^python@3.8.5%gcc@10.1.0`

`spack install strelka^python@2.7.18` works as intended.

### General information
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

I'll submit the PR within the hour.

--- a/var/spack/repos/builtin/packages/strelka/package.py
+++ b/var/spack/repos/builtin/packages/strelka/package.py
@@ -15,7 +15,7 @@ class Strelka(CMakePackage):
 
     version('2.8.2', sha256='27415f7c14f92e0a6b80416283a0707daed121b8a3854196872981d132f1496b')
 
-    depends_on('python@2.4:')
+    depends_on('python@2.4:2.7')
     depends_on('zlib')
     depends_on('bzip2')
     depends_on('cmake@2.8.5:')

